### PR TITLE
feat: add consumption api

### PR DESCRIPTION
**Pull request by bdraco**
- [x] R3.9.36 (a1a93f) 192.168.106.159 (89 nem)
- [x] D3.17.3 (0e8c7a) 192.168.121.200 (81 nem)
- [x] D5.0.62 (5e57a9) 192.168.111.40 (12 nem)
- [x] D7.3.130 (91f206) 192.168.106.37 (89 nem plus)

This should get us to parity with the old lib

### CODE REVIEW

**Review by cgarwood — APPROVED** [review]

**Comment by bdraco** [comment]
I'll start working on the ha integration migration tomorrow and do any followup changes if the data fluxes in any way

File: src/pyenphase/envoy.py [hunk 1]
```diff
@@ -58,9 +59,12 @@ def create_no_verify_ssl_context() -> ssl.SSLContext:
 
 class SupportedFeatures(enum.IntFlag):
     INVERTERS = 1
-    DRY_CONTACTS = 2
-    ENCHARGE = 4
-    ENPOWER = 8
+    METERING = 2
+    TOTAL_CONSUMPTION = 4
+    NET_CONSUMPTION = 8
+    DRY_CONTACTS = 16
```
Comment (joostlek) [inline src/pyenphase/envoy.py:65]: What's the reason dry contacts changed from 2 to 16? Like is there a technical reason for it? I'm curious

File: src/pyenphase/envoy.py [hunk 2]
```diff
@@ -168,18 +174,42 @@ def serial_number(self) -> str | None:
...
+        try:
+            production_json: dict[str, Any] = await self.request(URL_PRODUCTION)
+        except (json.JSONDecodeError, httpx.HTTPError) as e:
+            _LOGGER.debug("Production endpoint not found at %s: %s", URL_PRODUCTION, e)
+        else:
+            production: list[dict[str, str | float | int]] | None = production_json.get(
+                "production"
+            )
+            if production:
+                for type_ in production:
```
Comment (joostlek) [inline src/pyenphase/envoy.py:186]: What's the reason to pick `type_` over another variable name?

Comment (bdraco) [reply src/pyenphase/envoy.py:186]: type is reserved name in python

File: src/pyenphase/envoy.py (hunk 1 above)
Comment (bdraco) [reply src/pyenphase/envoy.py:65]: These could be auto as well. I was moving more likely features lower in the list for historical reasons (when we come look at this 10 years from now). It has no actual impact

**Pull request merged**